ext/src/ruby_api/engine.rs

```diff
@@ -26,11 +46,55 @@ impl Engine {
             None => EngineImpl::default(),
         };
 
-        Ok(Self { inner })
+        Ok(Self {
+            inner,
+            timer_task: Default::default(),
+        })
     }
 
-    pub fn get(&self) -> &EngineImpl {
-        &self.inner
+    /// @yard
+    /// Starts a timer that will increment the engine's epoch every +milliseconds+.
+    /// Waits +milliseconds+ before incrementing for the first time.
+    ///
+    /// If a prior timer was started, it will be stopped.
+    /// @def start_epoch_interval(milliseconds)
+    /// @param milliseconds [Integer]
+    /// @return [nil]
+    pub fn start_epoch_interval(&self, milliseconds: u64) {
+        self.stop_epoch_interval();
+        let engine = self.inner.clone();
+
+        let handle = TOKIO_RT.spawn(async move {
+            let tick_every = time::Duration::from_millis(milliseconds);
+            let mut interval = Interval::platform_new(tick_every);
+
+            loop {
+                interval.wait().await;
+                engine.increment_epoch();
+            }
+        });
+
+        *self.timer_task.borrow_mut() = Some(handle);
+    }
+
+    /// @yard
+    /// Stops a previously started timer with {#start_epoch_interval}.
+    /// Does nothing if there is no running timer.
+    /// @return [nil]
+    pub fn stop_epoch_interval(&self) {
+        if let Some(handle) = self.timer_task.take() {
+            handle.abort();
```

Comment:
Should we block until the handle `is_cancelled?` I'm wondering what the implications might be otherwise... i.e. if this sequence of events happens, are we safe?

1. Engine is dropped
2. `stop_interval` is called, but there is still a task running
3. Ruby decides to GC the engine
4. Timer calls `increment_epoch`

- I _assume_ we are OK because `Engine` is behind some type of refcell thingy
- But what about our `Store`? Does it being `UnsafeCell` any implications?